lib/ring_logger/client.ex

```diff
@@ -226,6 +234,22 @@ defmodule RingLogger.Client do
     end
   end
 
+  def handle_call(:count_next, _from, state) do
+    case Server.get(state.index, 0) do
+      [] ->
+        # No messages
+        {:reply, 0, state}
+
+      messages ->
+        count =
+          messages
+          |> Enum.filter(&should_print?(&1, state))
+          |> Enum.count()
```

Comment:
```suggestion
          |> Enum.count(&should_print?(&1, state))
```
Try this to avoid creating the big temporary list.